dask/array/core.py

```diff
@@ -569,18 +569,23 @@ def store(sources, targets, **kwargs):
     get(dsk, keys, **kwargs)
 
 
-def blockdims_from_blockshape(shape, blockshape):
+def blockdims_from_blockshape(shape, chunks):
     """
 
     >>> blockdims_from_blockshape((10, 10), (4, 3))
     ((4, 4, 2), (3, 3, 3, 1))
     """
-    if blockshape is None:
+    if chunks is None:
         raise TypeError("Must supply chunks= keyword argument")
     if shape is None:
         raise TypeError("Must supply shape= keyword argument")
+    if not all(map(lambda x: is_integer(x), shape + chunks)):
```

Comment:
`shape` and `chunks` can possibly be different types (`list` and `tuple`) so I have to check them separately, since `list + tuple` fails. That is what I get for trying to be clever.